lib/screenplay/config/permanent_config.ex

```diff
@@ -175,15 +175,33 @@ defmodule Screenplay.Config.PermanentConfig do
     %PendingConfig{screens: existing_screens} = PendingConfig.from_json(deserialized)
     %Config{screens: published_screens} = Config.from_json(published_config_deserialized)
 
-    all_screen_ids =
+    {new_screen_ids, changed_screen_ids, deleted_screen_ids} =
       Enum.flat_map(places_and_screens, fn {_place_id,
                                             %{
                                               "new_pending_screens" => new_screens,
                                               "updated_pending_screens" => updated_pending_screens
                                             }} ->
-        Enum.map(new_screens, fn %{"new_id" => new_id} -> new_id end) ++
-          get_updated_pending_screen_ids(updated_pending_screens)
-      end) ++ Map.keys(existing_screens) ++ Map.keys(published_screens)
+        new_screens ++ updated_pending_screens
+      end)
+      |> Enum.reduce({[], [], []}, fn screen,
+                                      {new_screen_ids, changed_screen_ids, deleted_screen_ids} ->
+        cond do
+          screen["is_deleted"] ->
+            {new_screen_ids, changed_screen_ids, [screen["screen_id"] | deleted_screen_ids]}
+
+          screen["new_id"] != nil and screen["screen_id"] == nil ->
+            {[screen["new_id"] | new_screen_ids], changed_screen_ids, deleted_screen_ids}
+
+          # screen is existing but had its ID changed.
+          true ->
```

Comment:
What about screens that are being updated, but without an id change? Do they come with a "new_id" value, such that this clause would handle them?

Reply:
Hmm that's a good point. They do not have a `new_id` value so would always hit the `true` condition here. But they honestly could be filtered out of any of the three lists here because they will exist in the `Map.keys(existing_screens)` call a few lines down. I should probably have the `true` condition return the `acc` as-is and create one more condition for if `new_id` and `screen_id` both have non-nil values. Then the `cond` will cover all of our bases:
1. deleted screens
2. new screens
3. existing screens that have a new ID
4. any other screen that we can ignore and will be in `existing_screens`.